### DISCUSSION TITLE
programmatically add a slug at the end of the route url

### DISCUSSION THREAD

**Discussion by dxgi** [nuxt/nuxt#17596] [Questions]
`
products/:id/:title
`

is there anyway to "programmatically" add the title if it doesn't exist?

**Comment by TheAlexLichter** [comment]
You mean if someone accesses `products/1` instead of `products/1/my-product`?

I think you can use a route middleware to redirect the first URL to the second and "generate" the slug based on an API call or some other logic.

**Reply by dxgi** [reply]
wouldn't that kind of waste the bandwidth? Since I have to make a request and check if that actual id exists then redirect them to the id with the slug in it which would create another request if it's valid

**Reply by TheAlexLichter** [reply]
Indeed, that would be one downside (if you rely on an API call to generate the slugs). Another alternative would be to dertermine all id/slug combination beforehand and pass them along (or even just create a redirect table wherever you host your application). This would reduce the amount of calls but isn't as flexible

**Reply by dxgi** [reply]
I see, Is there actually no real way of programmatically editing the route parameters?

**Reply by TheAlexLichter** [reply]
As I highlighted in the idea in my original post, there definitely are ways to change the route parameter *programmatically*. But **how** this can be done (e.g. with/without an API call) is up to you and your needs. 

If you know how the title parameter should look like, you can use a redirect (either "on the fly" or at build time) to correctly route the title-less URL to the URL with title.
